zipline/pipeline/term.py

```diff
@@ -516,6 +585,31 @@ def postprocess(self, data):
         """
         return data
 
+    def _downsampled_type(self):
+        """
+        The expression type to return from self.downsample().
+        """
+        raise NotImplementedError(
+            "downsampling is not yet implemented "
+            "for instances of %s." % type(self).__name__
+        )
+
+    def downsample(self, frequency):
+        """
+        Make a term that computes from ``self`` at lower-than-daily frequency.
+
+        Parameters
+        ----------
+        frequency : str, {'Y', 'Q', 'M', 'W'}
+            A string indicating the desired sampling rate.
+            'Y' -> sample on the first trading day of each calendar year
```

Comment:
My only objection to this is that pandas inexplicably users `A` and `AS` for yearly instead of `Y` and `YS`, which I have never once successfully remembered.